File: LAB3/2024-1/LAB3-2024_1-P4/MedicalSoftDBManager/src/com/medicalsoft/config/DBManager.java
package com.medicalsoft.config;

import java.sql.DriverManager;
import java.sql.Connection;

public class DBManager {

    private Connection con;
    private static DBManager dbManager;
    private static String url = "jdbc:mysql://prog3-labs-1inf30.c8ttwc6gkvxy.us-east-1.rds.amazonaws.com/laboratorio4";
    private static String username = "admin";
    private static String password = "labs1inf3020241";

	public static String getUrl() {
		return url;
	}

	public static String getUsername() {
		return username;
	}

	public static String getPassword() {
		return password;
	}

    public Connection getConnection(){
        try{
            Class.forName("com.mysql.cj.jdbc.Driver");
            con = DriverManager.getConnection(url, username, password);
        }catch(Exception ex){
            System.out.println(ex.getMessage());
        }
        return con;
    }

    public static DBManager getInstance(){
        if(dbManager == null){
            createInstance();
        }
        return dbManager;
    }

    private synchronized static void createInstance(){
        if(dbManager == null){
            dbManager = new DBManager();
        }
    }
}

File: LAB4/2024-2/EventMasterSoft/EventMasterSoftDBManager/DBManager.cs
using MySql.Data.MySqlClient;
using System;
using System.Collections.Generic;
using System.Data.SqlClient;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace EventMasterSoftDBManager
{
    public class DBManager
    {
        private static DBManager dbManager = null;
        private MySqlConnection con;
        private static string url = "server=labs-1inf30-prog3-20242.c8ttwc6gkvxy.us-east-1.rds.amazonaws.com;";
        private static string username = "user=admin;";
        private static string password = "password=prog320242labs;";
        private static string bd = "database=lab06;";
        private static string port = "port=3306;";

        private DBManager() { }
        public static DBManager Instance
        {
            get
            {
                if (dbManager == null)
                    createInstance();
                return dbManager;
            }
        }

        private static void createInstance()
        {
            if (dbManager == null)
                dbManager = new DBManager();
        }

        public MySqlConnection Connection
        {
            get
            {
                string cadena = url + username + password + port + bd;
                con = new MySqlConnection(cadena);
                return con;
            }
        }
    }
}
